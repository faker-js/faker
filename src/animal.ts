import type { Faker } from '.';

export class Animal {
  constructor(private readonly faker: Faker) {
    // Bind `this` so namespaced is working correctly
    for (const name of Object.getOwnPropertyNames(Animal.prototype)) {
      if (name === 'constructor' || typeof this[name] !== 'function') {
        continue;
      }
      this[name] = this[name].bind(this);
    }
  }

  /**
   * Returns a random dog breed.
   *
   */
  dog(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.dog);
  }

  /**
   * Returns a random cat breed.
   *
   */
  cat(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.cat);
  }

  /**
   * Returns a random snake species.
   *
   */
  snake(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.snake);
  }

  /**
   * Returns a random bear species.
   *
   */
  bear(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.bear);
  }

  /**
   * Returns a random lion species.
   *
   */
  lion(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.lion);
  }

  /**
   * Returns a random cetacean species.
   *
   * @method faker.animal.cetacean
   */
  cetacean(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.animal.cetacean
    );
  }

  /**
   * Returns a random horse breed.
   *
   */
  horse(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.horse);
  }

  /**
   * Returns a random bird species.
   *
   */
  bird(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.bird);
  }

  /**
   * Returns a random cow species.
   *
   */
  cow(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.cow);
  }

  /**
   * Returns a random fish species.
   *
   */
  fish(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.fish);
  }

  /**
   * Returns a random crocodilian species.
   *
   */
  crocodilia(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.animal.crocodilia
    );
  }

  /**
   * Returns a random insect species.
   *
   */
  insect(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.insect);
  }

  /**
   * Returns a random rabbit species
   *
   */
  rabbit(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.rabbit);
  }

  /**
   * Returns a random animal type.
   *
   */
  type(): string {
    return this.faker.random.arrayElement(this.faker.definitions.animal.type);
  }
}
