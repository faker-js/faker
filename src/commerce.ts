import type { Faker } from '.';
import { deprecated } from './internal/deprecated';

/**
 * Module to generate commerce and product related entries.
 */
export class Commerce {
  constructor(private readonly faker: Faker) {
    // Bind `this` so namespaced is working correctly
    for (const name of Object.getOwnPropertyNames(Commerce.prototype)) {
      if (name === 'constructor' || typeof this[name] !== 'function') {
        continue;
      }
      this[name] = this[name].bind(this);
    }
  }

  /**
   * Returns a human readable color name.
   *
   * @example
   * faker.commerce.color() // 'red'
   *
   * @deprecated
   */
  color(): string {
    deprecated({
      deprecated: 'faker.commerce.color()',
      proposed: 'faker.color.human()',
      since: 'v7.0.0',
      until: 'v8.0.0',
    });
    return this.faker.color.human();
  }

  /**
   * Returns a department inside a shop.
   *
   * @example
   * faker.commerce.department() // 'Garden'
   */
  department(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.commerce.department
    );
  }

  /**
   * Return a percentage value in decimal format between 0 and 1
   * with percision of two decimal place.
   *
   * @param precision Precision of the generated number. Defaults to `0.01`.
   *
   * @example
   * this.getPercentage(0.1) // 0.4
   * this.getPercentage(0.01) // 0.36
   */
  percentage(precision: number): number {
    return this.faker.datatype.float({
      min: 0,
      max: 1,
      precision: precision || 0.01,
    });
  }

  /**
   * Generates a random descriptive product name.
   *
   * @example
   * faker.commerce.productName() // 'Incredible Soft Gloves'
   */
  productName(): string {
    return (
      this.faker.commerce.productAdjective() +
      ' ' +
      this.faker.commerce.productMaterial() +
      ' ' +
      this.faker.commerce.product()
    );
  }

  /**
   * Generates a price between min and max (inclusive).
   *
   * @param min The minimum price. Defaults to `1`.
   * @param max The maximum price. Defaults to `1000`.
   * @param dec The number of decimal places. Defaults to `2`.
   * @param symbol The currency value to use. Defaults to `''`.
   *
   * @example
   * faker.commerce.price() // 828.00
   * faker.commerce.price(100) // 904.00
   * faker.commerce.price(100, 200) // 154.00
   * faker.commerce.price(100, 200, 0) // 133
   * faker.commerce.price(100, 200, 0, '$') // $114
   */
  price(
    min: number = 1,
    max: number = 1000,
    dec: number = 2,
    symbol: string = ''
  ): string {
    if (min < 0 || max < 0) {
      return `${symbol}${0.0}`;
    }

    const randValue = this.faker.datatype.number({ max: max, min: min });

    return (
      symbol +
      (Math.round(randValue * Math.pow(10, dec)) / Math.pow(10, dec)).toFixed(
        dec
      )
    );
  }

  /**
   * Returns an adjective describing a product.
   *
   * @example
   * faker.commerce.productAdjective() // 'Handcrafted'
   */
  productAdjective(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.commerce.product_name.adjective
    );
  }

  /**
   * Returns a material of a product.
   *
   * @example
   * faker.commerce.productMaterial() // 'Rubber'
   */
  productMaterial(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.commerce.product_name.material
    );
  }

  /**
   * Returns a short product name.
   *
   * @example
   * faker.commerce.product() // 'Computer'
   */
  product(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.commerce.product_name.product
    );
  }

  /**
   * Returns a product description.
   *
   * @example
   * faker.commerce.productDescription() // 'Andy shoes are designed to keeping...'
   */
  productDescription(): string {
    return this.faker.random.arrayElement(
      this.faker.definitions.commerce.product_description
    );
  }
}
