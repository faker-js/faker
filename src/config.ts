/**
 * The possible configuration options, that can be set.
 * This type exists to be extended for plugins via type augmentation.
 *
 * The `@default` tag is used to indicate the default value, that should be used if, the config is absent.
 */
export interface FakerConfig {
  /**
   * The function used to generate the `refDate` date instance, if not provided as method param.
   * The function must return a new valid `Date` instance for every call.
   *
   * @see [Reproducible Results](https://fakerjs.dev/guide/usage.html#reproducible-results)
   * @see faker.seed(): For generating reproducible values.
   *
   * @since 9.0.0
   *
   * @default () => new Date()
   */
  defaultRefDate?: () => Date;
}
