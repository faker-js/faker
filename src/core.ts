import type { FakerConfig } from './config';
import type { LocaleDefinition } from './definitions';
import type { Randomizer } from './randomizer';

/**
 * The core grants access to the locale data, the randomizer and config settings.
 */
export interface FakerCore {
  /**
   * The locale data associated with this instance.
   *
   * Always present, but it might be empty if the locale data is not available.
   */
  readonly locale: LocaleDefinition;

  /**
   * The randomizer used to generate random values.
   */
  readonly randomizer: Randomizer;

  /**
   * The configuration settings used by this instance.
   */
  readonly config: FakerConfig;
}
