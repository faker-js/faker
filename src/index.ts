import { Faker } from './faker';
import allLocales from './locales';

export type {
  AnimalDefinitions,
  ColorDefinitions,
  CommerceDefinitions,
  CommerceProductNameDefinitions,
  CompanyDefinitions,
  DatabaseDefinitions,
  DateDefinitions,
  DateEntryDefinition,
  FinanceCurrencyEntryDefinitions,
  FinanceDefinitions,
  HackerDefinitions,
  InternetDefinitions,
  LocaleDefinition,
  /** @deprecated Use LocationDefinitions instead */
  LocationDefinitions as AddressDefinitions,
  LocationDefinitions,
  LoremDefinitions,
  MusicDefinitions,
  /** @deprecated Use PersonDefinitions instead */
  PersonDefinitions as NameDefinitions,
  PersonDefinitions,
  /** @deprecated Use PersonTitleDefinitions instead */
  PersonTitleDefinitions as NameTitleDefinitions,
  PersonTitleDefinitions,
  PhoneNumberDefinitions,
  ScienceDefinitions,
  SystemDefinitions,
  SystemMimeTypeEntryDefinitions,
  VehicleDefinitions,
  WordDefinitions,
} from './definitions';
export { FakerError } from './errors/faker-error';
export type { FakerOptions, UsableLocale, UsedLocales } from './faker';
export type { AnimalModule } from './modules/animal';
export type {
  Casing,
  ColorFormat,
  ColorModule,
  CSSFunction,
  CSSSpace,
  NumberColorFormat,
  StringColorFormat,
} from './modules/color';
export type { CommerceModule } from './modules/commerce';
export type { CompanyModule } from './modules/company';
export type { DatabaseModule } from './modules/database';
export type { DatatypeModule } from './modules/datatype';
export type { DateModule } from './modules/date';
export type { FinanceModule } from './modules/finance';
export type { GitModule } from './modules/git';
export type { HackerModule } from './modules/hacker';
export type { HelpersModule } from './modules/helpers';
export type { ImageModule } from './modules/image';
export type { InternetModule } from './modules/internet';
export type {
  /** @deprecated Use LocationModule instead */
  LocationModule as AddressModule,
  LocationModule,
} from './modules/location';
export type { LoremModule } from './modules/lorem';
export type { MusicModule } from './modules/music';
export { Sex } from './modules/person';
export type {
  /** @deprecated Use PersonModule instead */
  PersonModule as NameModule,
  PersonModule,
  SexType,
} from './modules/person';
export type { PhoneModule } from './modules/phone';
export type { RandomModule } from './modules/random';
export type { ChemicalElement, ScienceModule, Unit } from './modules/science';
export type { StringModule } from './modules/string';
export type { SystemModule } from './modules/system';
export type { VehicleModule } from './modules/vehicle';
export type { WordModule } from './modules/word';
export { Faker };

// since we are requiring the top level of faker, load all locales by default
export const faker: Faker = new Faker({
  locales: allLocales,
});
