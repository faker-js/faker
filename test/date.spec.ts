import { afterEach, describe, expect, it } from 'vitest';
import { faker } from '../src';
import { seededTests } from './support/seededRuns';

const converterMap = [
  (d: Date) => d,
  (d: Date) => d.toISOString(),
  (d: Date) => d.valueOf(),
];

const NON_SEEDED_BASED_RUN = 5;
const refDate = '2021-02-21T17:09:15.711Z';

describe('date', () => {
  afterEach(() => {
    faker.locale = 'en';
  });

  seededTests(faker, 'date', (t) => {
    t.describeEach(
      'past',
      'recent',
      'soon',
      'future'
    )((t) => {
      t.it('with only string refDate', undefined, refDate)
        .it('with only Date refDate', undefined, new Date(refDate))
        .it('with value', 10, refDate);
    });

    t.describeEach(
      'weekday',
      'month'
    )((t) => {
      t.it('noArgs')
        .it('with abbr = true', { abbr: true })
        .it('with context = true', { context: true })
        .it('with abbr = true and context = true', {
          abbr: true,
          context: true,
        });
    });

    t.describe('between', (t) => {
      t.it(
        'with string dates',
        '2021-02-21T17:09:15.711Z',
        '2021-04-21T17:11:17.711Z'
      ).it(
        'with Date dates',
        new Date('2021-02-21T17:09:15.711Z'),
        new Date('2021-04-21T17:11:17.711Z')
      );
    });

    t.describe('betweens', (t) => {
      t.it(
        'with string dates',
        '2021-02-21T17:09:15.711Z',
        '2021-04-21T17:11:17.711Z'
      )
        .it(
          'with Date dates',
          new Date('2021-02-21T17:09:15.711Z'),
          new Date('2021-04-21T17:11:17.711Z')
        )
        .it(
          'with string dates and count',
          '2021-02-21T17:09:15.711Z',
          '2021-04-21T17:11:17.711Z',
          5
        )
        .it(
          'with Date dates and count',
          new Date('2021-02-21T17:09:15.711Z'),
          new Date('2021-04-21T17:11:17.711Z'),
          5
        );
    });

    t.describe('birthdate', (t) => {
      t.it('with only refDate', { refDate })
        .it('with age mode and refDate', {
          mode: 'age',
          refDate,
        })
        .it('with age and refDate', {
          min: 40,
          max: 40,
          mode: 'age',
          refDate,
        })
        .it('with age range and refDate', {
          min: 20,
          max: 80,
          mode: 'age',
          refDate,
        })
        .it('with year mode and refDate', {
          mode: 'year',
          refDate,
        })
        .it('with year and refDate', {
          min: 2000,
          max: 2000,
          mode: 'age',
          refDate,
        })
        .it('with year range and refDate', {
          min: 1900,
          max: 2000,
          mode: 'age',
          refDate,
        });
    });
  });

  describe(`random seeded tests for seed ${faker.seed()}`, () => {
    for (let i = 1; i <= NON_SEEDED_BASED_RUN; i++) {
      describe('past()', () => {
        it('should return a date 5 years in the past', () => {
          const today = new Date();
          const yearsAgo = new Date(today);
          yearsAgo.setFullYear(yearsAgo.getFullYear() - 5);

          const date = faker.date.past(5);

          expect(date).lessThan(today);
          expect(date).greaterThanOrEqual(yearsAgo);
        });

        it('should return a past date when years 0', () => {
          const refDate = new Date();
          const date = faker.date.past(0, refDate.toISOString());

          expect(date).lessThan(refDate);
        });

        it.each(converterMap)(
          'should return a past date relative to given refDate',
          (converter) => {
            const refDate = new Date();
            refDate.setFullYear(refDate.getFullYear() + 5);

            const date = faker.date.past(5, converter(refDate));

            expect(date).lessThan(refDate);
            expect(date).greaterThan(new Date());
          }
        );
      });

      describe('future()', () => {
        it('should return a date 75 years into the future', () => {
          const date = faker.date.future(75);

          expect(date).greaterThan(new Date());
        });

        it('should return a future date when years 0', () => {
          const refDate = new Date();
          const date = faker.date.future(0, refDate.toISOString());

          expect(date).greaterThan(refDate); // date should be after the date given
        });

        it.each(converterMap)(
          'should return a date 75 years after the date given',
          (converter) => {
            const refDate = new Date(1880, 11, 9, 10, 0, 0, 0); // set the date beyond the usual calculation (to make sure this is working correctly)

            const date = faker.date.future(75, converter(refDate));

            // date should be after the date given, but before the current time
            expect(date).greaterThan(refDate);
            expect(date).lessThan(new Date());
          }
        );
      });

      describe('between()', () => {
        it.each(converterMap)(
          'should return a random date between the dates given',
          (converter) => {
            const from = new Date(1990, 5, 7, 9, 11, 0, 0);
            const to = new Date(2000, 6, 8, 10, 12, 0, 0);

            const date = faker.date.between(converter(from), converter(to));

            expect(date).greaterThan(from);
            expect(date).lessThan(to);
          }
        );
      });

      describe('betweens()', () => {
        it.each(converterMap)(
          'should return an array of 3 dates ( by default ) of sorted randoms dates between the dates given',
          (converter) => {
            const from = new Date(1990, 5, 7, 9, 11, 0, 0);
            const to = new Date(2000, 6, 8, 10, 12, 0, 0);

            const dates = faker.date.betweens(converter(from), converter(to));

            expect(dates[0]).greaterThan(from);
            expect(dates[0]).lessThan(to);
            expect(dates[1]).greaterThan(dates[0]);
            expect(dates[2]).greaterThan(dates[1]);
          }
        );
      });

      describe('recent()', () => {
        it('should return a date N days from the recent past', () => {
          const date = faker.date.recent(30);

          expect(date).lessThanOrEqual(new Date());
        });

        it.each(converterMap)(
          'should return a date N days from the recent past, starting from refDate',
          (converter) => {
            const days = 30;
            const refDate = new Date(2120, 11, 9, 10, 0, 0, 0); // set the date beyond the usual calculation (to make sure this is working correctly)

            const lowerBound = new Date(
              refDate.getTime() - days * 24 * 60 * 60 * 1000
            );

            const date = faker.date.recent(days, converter(refDate));

            expect(
              lowerBound,
              '`recent()` date should not be further back than `n` days ago'
            ).lessThanOrEqual(date);
            expect(
              date,
              '`recent()` date should not be ahead of the starting date reference'
            ).lessThanOrEqual(refDate);
          }
        );
      });

      describe('soon()', () => {
        it('should return a date N days into the future', () => {
          const date = faker.date.soon(30);

          expect(date).greaterThanOrEqual(new Date());
        });

        it.each(converterMap)(
          'should return a date N days from the recent future, starting from refDate',
          (converter) => {
            const days = 30;
            const refDate = new Date(1880, 11, 9, 10, 0, 0, 0); // set the date beyond the usual calculation (to make sure this is working correctly)

            const upperBound = new Date(
              refDate.getTime() + days * 24 * 60 * 60 * 1000
            );

            const date = faker.date.soon(days, converter(refDate));

            expect(
              date,
              '`soon()` date should not be further ahead than `n` days ago'
            ).lessThanOrEqual(upperBound);
            expect(
              refDate,
              '`soon()` date should not be behind the starting date reference'
            ).lessThanOrEqual(date);
          }
        );
      });

      describe('month()', () => {
        it('should return random value from date.month.wide array by default', () => {
          const month = faker.date.month();
          expect(faker.definitions.date.month.wide).toContain(month);
        });

        it('should return random value from date.month.wide_context array for context option', () => {
          const month = faker.date.month({ context: true });
          expect(faker.definitions.date.month.wide_context).toContain(month);
        });

        it('should return random value from date.month.abbr array for abbr option', () => {
          const month = faker.date.month({ abbr: true });
          expect(faker.definitions.date.month.abbr).toContain(month);
        });

        it('should return random value from date.month.abbr_context array for abbr and context option', () => {
          const month = faker.date.month({ abbr: true, context: true });
          expect(faker.definitions.date.month.abbr_context).toContain(month);
        });

        it('should return random value from date.month.wide array for context option when date.month.wide_context array is missing', () => {
          const backup_wide_context = faker.definitions.date.month.wide_context;
          faker.definitions.date.month.wide_context = undefined;

          const month = faker.date.month({ context: true });
          expect(faker.definitions.date.month.wide).toContain(month);

          faker.definitions.date.month.wide_context = backup_wide_context;
        });

        it('should return random value from date.month.abbr array for abbr and context option when date.month.abbr_context array is missing', () => {
          const backup_abbr_context = faker.definitions.date.month.abbr_context;
          faker.definitions.date.month.abbr_context = undefined;

          const month = faker.date.month({ abbr: true, context: true });
          expect(faker.definitions.date.month.abbr).toContain(month);

          faker.definitions.date.month.abbr_context = backup_abbr_context;
        });
      });

      describe('weekday()', () => {
        it('should return random value from date.weekday.wide array by default', () => {
          const weekday = faker.date.weekday();
          expect(faker.definitions.date.weekday.wide).toContain(weekday);
        });

        it('should return random value from date.weekday.wide_context array for context option', () => {
          const weekday = faker.date.weekday({ context: true });
          expect(faker.definitions.date.weekday.wide_context).toContain(
            weekday
          );
        });

        it('should return random value from date.weekday.abbr array for abbr option', () => {
          const weekday = faker.date.weekday({ abbr: true });
          expect(faker.definitions.date.weekday.abbr).toContain(weekday);
        });

        it('should return random value from date.weekday.abbr_context array for abbr and context option', () => {
          const weekday = faker.date.weekday({ abbr: true, context: true });
          expect(faker.definitions.date.weekday.abbr_context).toContain(
            weekday
          );
        });

        it('should return random value from date.weekday.wide array for context option when date.weekday.wide_context array is missing', () => {
          const backup_wide_context =
            faker.definitions.date.weekday.wide_context;
          faker.definitions.date.weekday.wide_context = undefined;

          const weekday = faker.date.weekday({ context: true });
          expect(faker.definitions.date.weekday.wide).toContain(weekday);

          faker.definitions.date.weekday.wide_context = backup_wide_context;
        });

        it('should return random value from date.weekday.abbr array for abbr and context option when date.weekday.abbr_context array is missing', () => {
          const backup_abbr_context =
            faker.definitions.date.weekday.abbr_context;
          faker.definitions.date.weekday.abbr_context = undefined;

          const weekday = faker.date.weekday({ abbr: true, context: true });
          expect(faker.definitions.date.weekday.abbr).toContain(weekday);

          faker.definitions.date.weekday.abbr_context = backup_abbr_context;
        });
      });

      describe('birthdate', () => {
        it('returns a random birthdate', () => {
          const birthdate = faker.date.birthdate();
          expect(birthdate).toBeInstanceOf(Date);
        });

        it('returns a random birthdate between two years', () => {
          const min = 1990;
          const max = 2000;

          const birthdate = faker.date.birthdate({ min, max, mode: 'year' });

          // birthdate is a date object
          expect(birthdate).toBeInstanceOf(Date);

          // Generated date is between min and max
          expect(birthdate.getUTCFullYear()).toBeGreaterThanOrEqual(min);
          expect(birthdate.getUTCFullYear()).toBeLessThanOrEqual(max);
        });

        it('returns a random birthdate between two ages', () => {
          const min = 4;
          const max = 5;

          const birthdate = faker.date.birthdate({ min, max, mode: 'age' });

          // birthdate is a date object
          expect(birthdate).toBeInstanceOf(Date);

          // Generated date is between min and max
          expect(birthdate.getUTCFullYear()).toBeGreaterThanOrEqual(
            new Date().getUTCFullYear() - max - 1
          );
          expect(birthdate.getUTCFullYear()).toBeLessThanOrEqual(
            new Date().getUTCFullYear() - min
          );
        });
      });
    }
  });
});
