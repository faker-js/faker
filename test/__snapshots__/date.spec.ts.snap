// Vitest Snapshot v1

exports[`date > 42 > between > with Date dates 1`] = `2021-03-15T19:30:57.091Z`;

exports[`date > 42 > between > with string dates 1`] = `2021-03-15T19:30:57.091Z`;

exports[`date > 42 > betweens > with Date dates 1`] = `
[
  2021-03-15T19:30:57.091Z,
  2021-04-09T17:05:10.406Z,
  2021-04-18T19:23:52.973Z,
]
`;

exports[`date > 42 > betweens > with Date dates and count 1`] = `
[
  2021-03-04T12:54:15.263Z,
  2021-03-15T19:30:57.091Z,
  2021-04-05T21:40:57.315Z,
  2021-04-09T17:05:10.406Z,
  2021-04-18T19:23:52.973Z,
]
`;

exports[`date > 42 > betweens > with string dates 1`] = `
[
  2021-03-15T19:30:57.091Z,
  2021-04-09T17:05:10.406Z,
  2021-04-18T19:23:52.973Z,
]
`;

exports[`date > 42 > betweens > with string dates and count 1`] = `
[
  2021-03-04T12:54:15.263Z,
  2021-03-15T19:30:57.091Z,
  2021-04-05T21:40:57.315Z,
  2021-04-09T17:05:10.406Z,
  2021-04-18T19:23:52.973Z,
]
`;

exports[`date > 42 > birthdate > with age and refDate 1`] = `1980-07-07T19:06:53.022Z`;

exports[`date > 42 > birthdate > with age mode and refDate 1`] = `1963-09-27T06:10:33.792Z`;

exports[`date > 42 > birthdate > with age range and refDate 1`] = `1962-12-27T20:13:59.702Z`;

exports[`date > 42 > birthdate > with only refDate 1`] = `1964-08-06T01:03:57.017Z`;

exports[`date > 42 > birthdate > with year and refDate 1`] = `0020-07-07T19:06:53.022Z`;

exports[`date > 42 > birthdate > with year mode and refDate 1`] = `1964-08-06T01:03:57.017Z`;

exports[`date > 42 > birthdate > with year range and refDate 1`] = `0057-12-20T11:59:23.890Z`;

exports[`date > 42 > future > with only Date refDate 1`] = `2021-07-08T10:07:33.381Z`;

exports[`date > 42 > future > with only string refDate 1`] = `2021-07-08T10:07:33.381Z`;

exports[`date > 42 > future > with value 1`] = `2024-11-19T18:52:06.785Z`;

exports[`date > 42 > month > noArgs 1`] = `"May"`;

exports[`date > 42 > month > with abbr = true 1`] = `"May"`;

exports[`date > 42 > month > with abbr = true and context = true 1`] = `"May"`;

exports[`date > 42 > month > with context = true 1`] = `"May"`;

exports[`date > 42 > past > with only Date refDate 1`] = `2020-10-08T00:10:58.041Z`;

exports[`date > 42 > past > with only string refDate 1`] = `2020-10-08T00:10:58.041Z`;

exports[`date > 42 > past > with value 1`] = `2017-05-26T15:26:24.637Z`;

exports[`date > 42 > recent > with only Date refDate 1`] = `2021-02-21T08:09:54.820Z`;

exports[`date > 42 > recent > with only string refDate 1`] = `2021-02-21T08:09:54.820Z`;

exports[`date > 42 > recent > with value 1`] = `2021-02-17T23:15:52.427Z`;

exports[`date > 42 > soon > with only Date refDate 1`] = `2021-02-22T02:08:36.602Z`;

exports[`date > 42 > soon > with only string refDate 1`] = `2021-02-22T02:08:36.602Z`;

exports[`date > 42 > soon > with value 1`] = `2021-02-25T11:02:38.995Z`;

exports[`date > 42 > weekday > noArgs 1`] = `"Tuesday"`;

exports[`date > 42 > weekday > with abbr = true 1`] = `"Tue"`;

exports[`date > 42 > weekday > with abbr = true and context = true 1`] = `"Tue"`;

exports[`date > 42 > weekday > with context = true 1`] = `"Tuesday"`;

exports[`date > 1211 > between > with Date dates 1`] = `2021-04-17T11:58:13.327Z`;

exports[`date > 1211 > between > with string dates 1`] = `2021-04-17T11:58:13.327Z`;

exports[`date > 1211 > betweens > with Date dates 1`] = `
[
  2021-03-20T19:08:07.621Z,
  2021-04-15T10:20:25.794Z,
  2021-04-17T11:58:13.327Z,
]
`;

exports[`date > 1211 > betweens > with Date dates and count 1`] = `
[
  2021-03-07T00:34:12.770Z,
  2021-03-20T19:08:07.621Z,
  2021-04-08T15:12:37.581Z,
  2021-04-15T10:20:25.794Z,
  2021-04-17T11:58:13.327Z,
]
`;

exports[`date > 1211 > betweens > with string dates 1`] = `
[
  2021-03-20T19:08:07.621Z,
  2021-04-15T10:20:25.794Z,
  2021-04-17T11:58:13.327Z,
]
`;

exports[`date > 1211 > betweens > with string dates and count 1`] = `
[
  2021-03-07T00:34:12.770Z,
  2021-03-20T19:08:07.621Z,
  2021-04-08T15:12:37.581Z,
  2021-04-15T10:20:25.794Z,
  2021-04-17T11:58:13.327Z,
]
`;

exports[`date > 1211 > birthdate > with age and refDate 1`] = `1981-01-26T13:16:31.421Z`;

exports[`date > 1211 > birthdate > with age mode and refDate 1`] = `1998-08-21T21:24:30.782Z`;

exports[`date > 1211 > birthdate > with age range and refDate 1`] = `1996-10-13T01:44:07.645Z`;

exports[`date > 1211 > birthdate > with only refDate 1`] = `1999-06-29T11:06:58.506Z`;

exports[`date > 1211 > birthdate > with year and refDate 1`] = `0021-01-26T13:16:31.421Z`;

exports[`date > 1211 > birthdate > with year mode and refDate 1`] = `1999-06-29T11:06:58.506Z`;

exports[`date > 1211 > birthdate > with year range and refDate 1`] = `0113-12-03T19:45:27.654Z`;

exports[`date > 1211 > future > with only Date refDate 1`] = `2022-01-26T14:59:27.351Z`;

exports[`date > 1211 > future > with only string refDate 1`] = `2022-01-26T14:59:27.351Z`;

exports[`date > 1211 > future > with value 1`] = `2030-06-03T19:31:11.467Z`;

exports[`date > 1211 > month > noArgs 1`] = `"December"`;

exports[`date > 1211 > month > with abbr = true 1`] = `"Dec"`;

exports[`date > 1211 > month > with abbr = true and context = true 1`] = `"Dec"`;

exports[`date > 1211 > month > with context = true 1`] = `"December"`;

exports[`date > 1211 > past > with only Date refDate 1`] = `2020-03-19T19:19:04.071Z`;

exports[`date > 1211 > past > with only string refDate 1`] = `2020-03-19T19:19:04.071Z`;

exports[`date > 1211 > past > with value 1`] = `2011-11-12T14:47:19.955Z`;

exports[`date > 1211 > recent > with only Date refDate 1`] = `2021-02-20T18:52:11.498Z`;

exports[`date > 1211 > recent > with only string refDate 1`] = `2021-02-20T18:52:11.498Z`;

exports[`date > 1211 > recent > with value 1`] = `2021-02-12T10:18:34.226Z`;

exports[`date > 1211 > soon > with only Date refDate 1`] = `2021-02-22T15:26:19.924Z`;

exports[`date > 1211 > soon > with only string refDate 1`] = `2021-02-22T15:26:19.924Z`;

exports[`date > 1211 > soon > with value 1`] = `2021-03-02T23:59:57.196Z`;

exports[`date > 1211 > weekday > noArgs 1`] = `"Saturday"`;

exports[`date > 1211 > weekday > with abbr = true 1`] = `"Sat"`;

exports[`date > 1211 > weekday > with abbr = true and context = true 1`] = `"Sat"`;

exports[`date > 1211 > weekday > with context = true 1`] = `"Saturday"`;

exports[`date > 1337 > between > with Date dates 1`] = `2021-03-09T04:11:24.667Z`;

exports[`date > 1337 > between > with string dates 1`] = `2021-03-09T04:11:24.667Z`;

exports[`date > 1337 > betweens > with Date dates 1`] = `
[
  2021-03-03T01:51:22.512Z,
  2021-03-09T04:11:24.667Z,
  2021-03-26T18:53:00.564Z,
]
`;

exports[`date > 1337 > betweens > with Date dates and count 1`] = `
[
  2021-03-03T01:51:22.512Z,
  2021-03-06T06:11:08.446Z,
  2021-03-09T04:11:24.667Z,
  2021-03-10T02:59:27.376Z,
  2021-03-26T18:53:00.564Z,
]
`;

exports[`date > 1337 > betweens > with string dates 1`] = `
[
  2021-03-03T01:51:22.512Z,
  2021-03-09T04:11:24.667Z,
  2021-03-26T18:53:00.564Z,
]
`;

exports[`date > 1337 > betweens > with string dates and count 1`] = `
[
  2021-03-03T01:51:22.512Z,
  2021-03-06T06:11:08.446Z,
  2021-03-09T04:11:24.667Z,
  2021-03-10T02:59:27.376Z,
  2021-03-26T18:53:00.564Z,
]
`;

exports[`date > 1337 > birthdate > with age and refDate 1`] = `1980-05-27T14:46:44.831Z`;

exports[`date > 1337 > birthdate > with age mode and refDate 1`] = `1956-08-25T03:57:00.497Z`;

exports[`date > 1337 > birthdate > with age range and refDate 1`] = `1956-02-15T21:16:40.120Z`;

exports[`date > 1337 > birthdate > with only refDate 1`] = `1957-07-05T09:38:29.057Z`;

exports[`date > 1337 > birthdate > with year and refDate 1`] = `0020-05-27T14:46:44.831Z`;

exports[`date > 1337 > birthdate > with year mode and refDate 1`] = `1957-07-05T09:38:29.057Z`;

exports[`date > 1337 > birthdate > with year range and refDate 1`] = `0046-08-09T19:19:18.047Z`;

exports[`date > 1337 > future > with only Date refDate 1`] = `2021-05-28T08:29:26.637Z`;

exports[`date > 1337 > future > with only string refDate 1`] = `2021-05-28T08:29:26.637Z`;

exports[`date > 1337 > future > with value 1`] = `2023-10-06T02:30:58.333Z`;

exports[`date > 1337 > month > noArgs 1`] = `"April"`;

exports[`date > 1337 > month > with abbr = true 1`] = `"Apr"`;

exports[`date > 1337 > month > with abbr = true and context = true 1`] = `"Apr"`;

exports[`date > 1337 > month > with context = true 1`] = `"April"`;

exports[`date > 1337 > past > with only Date refDate 1`] = `2020-11-18T01:49:04.785Z`;

exports[`date > 1337 > past > with only string refDate 1`] = `2020-11-18T01:49:04.785Z`;

exports[`date > 1337 > past > with value 1`] = `2018-07-11T07:47:33.089Z`;

exports[`date > 1337 > recent > with only Date refDate 1`] = `2021-02-21T10:51:56.041Z`;

exports[`date > 1337 > recent > with only string refDate 1`] = `2021-02-21T10:51:56.041Z`;

exports[`date > 1337 > recent > with value 1`] = `2021-02-19T02:16:05.653Z`;

exports[`date > 1337 > soon > with only Date refDate 1`] = `2021-02-21T23:26:35.381Z`;

exports[`date > 1337 > soon > with only string refDate 1`] = `2021-02-21T23:26:35.381Z`;

exports[`date > 1337 > soon > with value 1`] = `2021-02-24T08:02:25.769Z`;

exports[`date > 1337 > weekday > noArgs 1`] = `"Monday"`;

exports[`date > 1337 > weekday > with abbr = true 1`] = `"Mon"`;

exports[`date > 1337 > weekday > with abbr = true and context = true 1`] = `"Mon"`;

exports[`date > 1337 > weekday > with context = true 1`] = `"Monday"`;
